src/browser/modules/Stream/Auth/ConnectForm.tsx

```diff
@@ -0,0 +1,235 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, { useState, useEffect } from 'react'
+import Render from 'browser-components/Render'
+import { FormButton } from 'browser-components/buttons'
+import {
+  StyledConnectionForm,
+  StyledConnectionTextInput,
+  StyledConnectionSelect,
+  StyledConnectionLabel,
+  StyledConnectionFormEntry,
+  StyledSegment,
+  StyledBoltUrlHintText,
+  Flex,
+  FlexSpaceBetween,
+  FlexEnd
+} from './styled'
+import { NATIVE, NO_AUTH } from 'services/bolt/boltHelpers'
+import { toKeyString } from 'services/utils'
+import { stripScheme, getScheme } from 'services/boltscheme.utils'
+
+type AuthenticationMethod = typeof NATIVE | typeof NO_AUTH
+const authMethods: AuthenticationMethod[] = [NATIVE, NO_AUTH]
+const readableauthenticationMethods: Record<AuthenticationMethod, string> = {
+  [NATIVE]: 'Username / Password',
+  [NO_AUTH]: 'No authentication'
+}
+
+interface ConnectFormProps {
+  allowedSchemes: string[]
+  authenticationMethod: string
+  host: string
+  onAuthenticationMethodChange: () => void
+  onConnectClick: (doneFn?: () => void) => void
+  onHostChange: (fallbackScheme: string, newHost: string) => void
+  onUsernameChange: () => void
+  onPasswordChange: () => void
+  onDatabaseChange: () => void
+  database: string
+  password: string
+  username: string
+  used: boolean
+  showDbField: boolean
+}
+
+export default function ConnectForm(props: ConnectFormProps): JSX.Element {
+  const [connecting, setConnecting] = useState(false)
+  const [scheme, setScheme] = useState(
+    props.allowedSchemes ? `${getScheme(props.host)}://` : ''
+  )
+
+  useEffect(() => {
+    if (props.allowedSchemes) {
+      return setScheme(`${getScheme(props.host)}://`)
+    }
+    setScheme('')
+  }, [props.host, props.allowedSchemes])
+
+  // Add scheme when copying text from bolt url field
+  const onCopyBoltUrl = (e: React.ClipboardEvent<HTMLInputElement>) => {
+    const selection = document.getSelection()
+    if (!selection) {
+      return
+    }
+    e.preventDefault()
+    e.stopPropagation()
+    let val = selection.toString()
+    if (scheme) {
+      val = `${scheme}${stripScheme(val)}`
+    }
+    e.clipboardData?.setData('text', val)
+  }
+
+  const onHostChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const val = e.target.value
+    props.onHostChange(getScheme(scheme), val)
+  }
+
+  const onSchemeChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    const val = e.target.value
+    props.onHostChange(getScheme(val), stripScheme(props.host))
+  }
+
+  const onConnectClick = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault()
+    setConnecting(true)
+    props.onConnectClick(() => setConnecting(false))
+  }
+
+  return (
+    <StyledConnectionForm onSubmit={onConnectClick}>
+      <FlexSpaceBetween>
```

Comment:
Not a 100% happy with this styling, it feels more convoluted than it needs to be. I opted to make theses simple Flex components rather than inline style even if they are very simple, for consistency's sake